### ISSUE TITLE
Unsupported argument allow_blob_public_access in azurerm_storage_account with v3.0.0

### ISSUE THREAD

**Issue by stdtom** [hashicorp/terraform-provider-azurerm#16024]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v1.1.7
on darwin_amd64
+ provider registry.terraform.io/hashicorp/azurerm v3.0.0
```

### Affected Resource(s)

* `azurerm_storage_account`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      version = "= 3.0.0"
    }
  }
}

provider "azurerm" {
  subscription_id = var.subscription_id
  tenant_id       = var.tenant_id

  features {}
}

resource "azurerm_storage_account" "storage" {
  name                      = "mystorage"
  resource_group_name       = "myresourcegroup"
  location                  = "eastus"
  account_kind              = "StorageV2"
  account_tier              = "Standard"
  account_replication_type  = "LRS"
  enable_https_traffic_only = true
  allow_blob_public_access  = false
  min_tls_version           = "TLS1_2"
}
```

### Expected Behaviour

The resource should be created like it is done with version 2.99.0

### Actual Behaviour

When running `terraform apply`, terraform exitst with the following error message:

```
╷
│ Error: Unsupported argument
│ 
│   on main.tf line xx, in resource "azurerm_storage_account" "storage":
│   33:   allow_blob_public_access  = false
│ 
│ An argument named "allow_blob_public_access" is not expected here.
╵
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by tombuildsstuff** [comment]
hi @stdtom 

Thanks for opening this issue.

As 3.0 is a major release a complete list of changes can be found [in the 3.0 upgrade guide](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide) - which is [linked from the changelog since this is a major version](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/CHANGELOG.md#300-march-24-2022).

In this case this field has been renamed to `allow_nested_items_to_be_public`, however there maybe other changes you need to make to your Terraform Configuration to be compatible with v3.0 - we'd recommend consulting the upgrade guide.

We'd recommend pinning the version of the Azure Provider being used, for example to pin to `3.0.0`:

```hcl
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "=3.0.0"
    }
  }
}
```

You can also pin to the last 2.x release temporarily as shown below, however note that any future enhancements will ship in 3.x:

```hcl
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "=2.99.0"
    }
  }
}
```

Thanks!

**Comment by stdtom** [comment]
Hi @tombuildsstuff,
thanks for the swift reply and your hints. I have to admit that I missed the upgrade guide. 🙈 🙉 🙊 

Would it be possible to mention this in CHANGELOG.md?

Also, please be informed that this change is not yet reflected in [the docs](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account) - although I cannot rule out that this is a CDN caching issue.

**Comment by katbyte** [comment]
@stdtom we missed updating it in the docs 😅  but that's been fixed by #16025 and a docs release will go out tomorrow to get it updated on the website.

What would you suggest we add to the changelog? an explicit note for this property in the notes section?
